Algorithm.Python/EmaCrossUniverseSelectionAlgorithm.py

```diff
@@ -32,76 +32,86 @@
 ### <meta name="tag" content="indicators" />
 ### <meta name="tag" content="universes" />
 ### <meta name="tag" content="coarse universes" />
-class EmaCrossUniverseSelectionAlgorithm(QCAlgorithm):
-
+class EmaCrossUniverse(QCAlgorithm):
     def Initialize(self):
-        '''Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.'''
-
-        self.SetStartDate(2010,1,1)  #Set Start Date
-        self.SetEndDate(2015,1,1)    #Set End Date
-        self.SetCash(100000)           #Set Strategy Cash
-
+        self.SetStartDate(2019, 12, 23)
+        self.SetEndDate(2020,1,15)
+        self.SetCash(100000)
         self.UniverseSettings.Resolution = Resolution.Daily
-        self.UniverseSettings.Leverage = 2
-
-        self.coarse_count = 10
-        self.averages = { };
-
-        # this add universe method accepts two parameters:
-        # - coarse selection function: accepts an IEnumerable<CoarseFundamental> and returns an IEnumerable<Symbol>
         self.AddUniverse(self.CoarseSelectionFunction)
-
-
-    # sort the data by daily dollar volume and take the top 'NumberOfSymbols'
-    def CoarseSelectionFunction(self, coarse):
-
-        # We are going to use a dictionary to refer the object that will keep the moving averages
-        for cf in coarse:
-            if cf.Symbol not in self.averages:
-                self.averages[cf.Symbol] = SymbolData(cf.Symbol)
-
-            # Updates the SymbolData object with current EOD price
-            avg = self.averages[cf.Symbol]
-            avg.update(cf.EndTime, cf.AdjustedPrice)
-
-        # Filter the values of the dict: we only want up-trending securities
-        values = list(filter(lambda x: x.is_uptrend, self.averages.values()))
-
-        # Sorts the values of the dict: we want those with greater difference between the moving averages
-        values.sort(key=lambda x: x.scale, reverse=True)
-
-        for x in values[:self.coarse_count]:
-            self.Log('symbol: ' + str(x.symbol.Value) + '  scale: ' + str(x.scale))
-
-        # we need to return only the symbol objects
-        return [ x.symbol for x in values[:self.coarse_count] ]
-
-    # this event fires whenever we have changes to our universe
+        self.averages = {}
+        
+    def CoarseSelectionFunction(self,coarse):
+        
+        sorted_by_volume = sorted([cf for cf in coarse if cf.Price > 10],
+            key = lambda cf: cf.DollarVolume, reverse=True)
+            
+        selected = {cf.Symbol: cf for cf in sorted_by_volume[:100]}
+        symbols = list(selected.keys())
+        
+        new_symbols = [s for s in symbols if s not in self.averages]
+        if new_symbols:
+            
+            # Get history for all new symbols. If empty, log and return the selected -- WOW!
```

Comment:
Please remove the "WOW!". We could add more comments to this method. Before line 74, state that we will use CoarseFundamental.AdjustedPrice for symbols that have been added before. Before line 77, state that we select symbols that are trending upward (fast > low).